tcms/testruns/views.py

```diff
@@ -63,55 +61,64 @@ def post(self, request):
                                                'category',
                                                'priority').order_by('pk')
 
-        if request.POST.get('POSTING_TO_CREATE'):
-            form = NewRunForm(request.POST)
-            form.populate(product_id=test_plan.product_id)
-
-            if form.is_valid():
-                # Process the data in form.cleaned_data
-                default_tester = form.cleaned_data['default_tester']
-
-                test_run = TestRun.objects.create(
-                    product_version=test_plan.product_version,
-                    stop_date=None,
-                    summary=form.cleaned_data.get('summary'),
-                    notes=form.cleaned_data.get('notes'),
-                    plan=test_plan,
-                    build=form.cleaned_data['build'],
-                    manager=form.cleaned_data['manager'],
-                    default_tester=default_tester,
-                )
-
-                loop = 1
-                for case in form.cleaned_data['case']:
-                    try:
-                        tcp = TestCasePlan.objects.get(plan=test_plan, case=case)
-                        sortkey = tcp.sortkey
-                    except ObjectDoesNotExist:
-                        sortkey = loop * 10
-
-                    test_run.add_case_run(case=case, sortkey=sortkey,
-                                          assignee=default_tester)
-                    loop += 1
-
-                return HttpResponseRedirect(
-                    reverse('testruns-get', args=[test_run.pk, ])
-                )
+        form = NewRunForm(initial={
+            'summary': 'Test run for %s' % test_plan.name,
+            'manager': test_plan.author.email,
+            'default_tester': request.user.email,
+            'notes': '',
+            'plan': test_plan
+        })
+        form.populate(test_plan)
 
-        else:
-            form = NewRunForm(initial={
-                'summary': 'Test run for %s' % test_plan.name,
-                'manager': test_plan.author.email,
-                'default_tester': request.user.email,
-                'notes': '',
-            })
-            form.populate(product_id=test_plan.product_id)
+        context_data = {
+            'test_cases': tcs_values,
+            'form': form,
+            'disabled_cases': get_disabled_test_cases_count(test_cases)
+        }
+        return render(request, self.template_name, context_data)
 
+    def post(self, request):
+        plan_id = request.POST.get('plan')
+        if not plan_id:
+            messages.add_message(request,
+                                 messages.ERROR,
+                                 _('Creating a TestRun requires a TestPlan, select one'))
+            return HttpResponseRedirect(reverse('plans-search'))
```

Comment:
This error checking should only be part of the get method & update line 98 in test_views.py as well. If something is missing the form will be invalid and it will contain all of the necessary error messages.